test/e2e/networking.go

```diff
@@ -121,8 +121,21 @@ var _ = Describe("Networking", func() {
 		filterNodes(nodes, func(node api.Node) bool {
 			return isNodeReadySetAsExpected(&node, true)
 		})
+
 		if len(nodes.Items) < 2 {
-			Failf("Less than two nodes were found Ready.")
+			// in general, the test requires two nodes. But for local development, often a one node cluster
+			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
+			// only in some cases
+			if providerIs("gce", "gke") {
+				Failf(fmt.Sprintf("The test requires two nodes on %s", testContext.Provider))
+			} else {
+				if len(nodes.Items) == 1 {
+					Logf("Only one ready node is detected. The test has limited scope is such setting. " +
+						"Rerun it with at least two nodes to get complete coverage.")
+				} else {
+					Failf("No Ready nodes found.")
+				}
+			}
```

Comment:
This could be unindented. https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow